src/quic.rs

```diff
@@ -228,9 +229,24 @@ impl QuicClient {
 
         tpus_info.extend(self.extra_tpu_forward.iter().cloned());
 
-        self.leaders_selector
-            .block_leaders(&mut tpus_info, &blocklist_keys)
-            .await;
+        let snapshot = self.policies.snapshot();
+
+        let before_policy_check_tpu_infos_count = tpus_info.len();
+        let tpus_info: Vec<TpuInfo> = tpus_info
+            .into_iter()
+            .filter(
+                |tpu_info| match snapshot.is_allowed(&policies, &tpu_info.leader) {
+                    Ok(allowed) => allowed,
+                    Err(_) => {
+                        shield_policies_not_found_inc();
+
+                        false
+                    }
+                },
+            )
+            .collect();
+
+        sts_tpu_denied_inc_by(before_policy_check_tpu_infos_count - tpus_info.len());
```

Comment:
In follow may consider storing metrics locally on the policy store and then scraping in reporting thread.